### DISCUSSION TITLE
How to have undo in zsh on macos?

### DISCUSSION THREAD

**Discussion by espentrydal** [wezterm/wezterm#4266] [Q&A] [answered]
Hi

I can't find how to have undo on the command line in zsh in macos. It works in other terminals (Terminal, Alacritty).

`bindkey` gives `"^_" undo`

First CTRL-_ reduced font size so I disabled it according to the instructions in [https://github.com/wez/wezterm/issues/1919](Readline keybinding for undo (CTRL + SHIFT + - ) is replaced by DecreaseFontSize ( CTRL + - ))

My config looks like this
```
config.keys = {
	{ key = "-", mods = "CTRL", action = wezterm.action.DisableDefaultAssignment },
	{ key = "-", mods = "SHIFT|CTRL", action = wezterm.action.DisableDefaultAssignment },
	{ key = "+", mods = "CTRL", action = wezterm.action.DisableDefaultAssignment },
	{ key = "+", mods = "SHIFT|CTRL", action = wezterm.action.DisableDefaultAssignment },
	{ key = "mapped:_", mods = "CTRL", action = wezterm.action.DisableDefaultAssignment },
	{ key = "mapped:_", mods = "SHIFT|CTRL", action = wezterm.action.DisableDefaultAssignment },
}
```
But when I press CTRL-_ (or SHIFT CTRL- -) I only get the character - on the line. I also tried to disable all keybindings but still no luck.

Anyone know how to make it work? I have a Norwegian keyboard, so - and _ is on the same key (_ is SHIFT- - ).

**Comment by wez** [comment]
Take a look at https://wezfurlong.org/wezterm/troubleshooting.html#debugging-keyboard-related-issues

**Comment by espentrydal** [comment, marked as answer]
Thanks! Putting `config.use_ime = false` in my config fixed it.
